mule-extensions-api/src/main/java/org/mule/runtime/extension/internal/notification/ExtensionAction.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.extension.internal.notification;
+
+import org.mule.runtime.api.notification.Notification.Action;
+
+/**
+ * {@link Action} produced by extensions.
+ */
+public class ExtensionAction implements Action {
```

Comment:
does this really need to be api? seems like the user will use the NotificationActionDefinition to fire the notification. This seems more like an implementation internal to me

Reply:
it's in an internal package 🤔

Reply:
precisely. The package is internal, thus it'd be great to be able to change this class without forcing a new api version. If you have a good reason for this to be here, then fine. If not, I think we should move it

Reply:
I'm not sure I understand where we could move it to. If anything I can rename this to DefaultAction since it does not really depend on extensions.